src/slowmokit/methods/linear_model/logistic_regression/logistic_regression.hpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file methods/linear_model/logistic_regression/logistic_regression.hpp
+ *
+ * The header file including the logistic regression algorithm
+ */
+
+#ifndef SLOWMOKIT_LOGISTIC_REGRESSION_HPP
+#define SLOWMOKIT_LOGISTIC_REGRESSION_HPP
+
+#include "../../../core.hpp"
+
+template<class T> class LogisticRegression
+{
+  private:
+  std::vector<std::vector<T>> beta; // Initializing beta i.e weights ->
+                                    // (output_classes * numbers_of_features)
+  /**
+   * @brief computes softmax of all x values
+   *
+   * @param x values for which we are computing softmax
+   * @return std::vector<double> softmax values
+   */
+  std::vector<double> softmax(std::vector<T> x);
+  /**
+   * @brief Cross Entropy
+   *
+   * @param y output values one hot encoded
+   * @param qi predicted values
+   * @return double cross entropy
+   */
+  double crossEntropy(std::vector<int> y, std::vector<T> qi);
+  /**
+   * @brief computes loss
+   *
+   * @param x training values
+   * @param beta weights
+   * @return double loss value
+   */
+  double EvalL(std::vector<std::vector<T>> x, std::vector<std::vector<int>> y,
+               std::vector<std::vector<T>> beta);
+  /**
+   * @brief Logistic regression
```

Comment:
Change it to `Mini-batch gradient descent`.
Also, add the batch-size parameter here